### DISCUSSION TITLE
Facing UTF-8 issue while running gauge scripts

### DISCUSSION THREAD

**Discussion by jeevan-ui** [getgauge/taiko#1869] [Development]
Unable to connect to plugin Html Report 4.0.12. rpc error: code = Internal desc = grpc: error while marshaling: string field contains invalid UTF-8

I am facing this error while running gauge scripts, kindly help me.

**Comment by sriv** [comment]
is there an example test suite that we can use to replicate this?

**Reply by jeevan-ui** [reply]
Hi Srikanth,

I may not have what you have test suite, but I hope the following content might give you a pretty good understanding of the issue.


Login.js

step("Sign in as GUEST from data file <table>", async function(table) {
    table.entries(function (entry) {
    write((`${entry["ID"]}`),into(textBox({id:'form--checkout_signin_new_user--field--NEW_USER_EMAIL_ADDRESS'})));
    });
    // await click("CONTINUE AS GUEST");
});

step("Continue As Guest", async function() {
    // await evaluate(link({"id":"checkout_signin_new_user-submit"}), ele => ele.click());
    await click($("#checkout_signin_new_user-submit"));
});

step("FIRSTNAME <table>", async function(table) {
    table.entries(function (entry) {
	write((`${entry["FIRSTNAME"]}`),into(textBox({id:'form--address_shipping--field--FIRST_NAME'})));
	
    });

});


Login.spec

* Sign in as GUEST from data file <table:D:\nameus.csv>
* Wait for 3 secs
* Continue As Guest

## Shipping details
* Wait for 3 secs
* FIRSTNAME <table:D:\nameus.csv>

Note: While I am running the spec script it is not able to retrieve the data from the excel file instead i am getting the error I have mentioned in the post.
Please help me on this.

**Reply by sriv** [reply]
I suspect that the csv file is saved with incorrect encoding. Can you confirm if the `nameus.csv` file is saved with utf-8 encoding?
